lib/routes/yamibo/utils.ts

```diff
@@ -0,0 +1,101 @@
+import { parseDate } from '@/utils/parse-date';
+import timezone from '@/utils/timezone';
+import ofetch from '@/utils/ofetch';
+import { config } from '@/config';
+import asyncPool from 'tiny-async-pool';
+import { JSDOM } from 'jsdom';
+
+export function getDate(date: string): Date {
+    return timezone(parseDate(date), 8);
+}
+
+export async function fetchThread(
+    tid: string,
+    options?: {
+        ordertype?: string;
+        _dsign?: string;
+    },
+    retry = 0
+): Promise<{
+    link: string;
+    data?: string;
+}> {
+    const { auth, salt } = config.yamibo;
+    const params = new URLSearchParams();
+    params.set('mod', 'viewthread');
+    params.set('tid', tid);
+    if (options?.ordertype) {
+        params.set('ordertype', options.ordertype);
+    }
+    if (options?._dsign) {
+        params.set('_dsign', options._dsign);
+    }
+    const link = `https://bbs.yamibo.com/forum.php?${params.toString()}`;
+
+    const headers: HeadersInit = {};
+
+    if (auth && salt) {
+        headers.cookie = `EeqY_2132_saltkey=${salt}; EeqY_2132_auth=${auth}`;
+    }
+
+    const data = await ofetch<string>(link, { headers });
+
+    // sometimes may trigger anti-crawling measures
+    if (data.startsWith('<script type="text/javascript">') && retry <= 3) {
```

Comment:
HTML doesn't start with `<script`

Reply:
Sometimes when accessing [bbs.yamibo.com](https://bbs.yamibo.com/), anti-crawling mechanisms are triggered, and the response returns a script segment instead of the complete HTML content. Inside the script, there's a URL concatenation that includes a _dsign parameter, and it redirects through the location object. You must carry the correct _dsign parameter to request the proper page content.

![CleanShot 2024-05-11 at 12 59 18@2x](https://github.com/DIYgod/RSSHub/assets/13167154/7792a02a-db6c-497a-8024-ba27c3aa2b49)

https://gist.github.com/KarasuShin/68e7d7dca5894c59c990de6668989aaa